src/test/java/rest/gorestcoIn/ApiRest.java

```diff
@@ -0,0 +1,67 @@
+package rest.gorestcoIn;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+import rest.gorestcoIn.utils.User;
+import rest.gorestcoIn.utils.helpermethods.ApiCall;
+
+public class ApiRest {
+    private final String endPoint = "users/";
+    private int userID;
+
+    private final Integer postStatusCode = 201;
+    private final Integer errorStatusCode = 404;
+
+    @BeforeClass
+    public void setup() {
+        RestAssured.baseURI = "https://gorest.co.in/";
+        RestAssured.basePath = "public-api/";
+    }
+
+    @Test
+    public void postUserTest() {
+
+        User user = new User();
+
+        Response responseForPost = ApiCall.post(user, endPoint);
+
+        Assert.assertEquals(responseForPost.path("code"), postStatusCode);
+
+        userID = responseForPost.path("data.id");
+
+        Response responseForGet = ApiCall.get(endPoint, userID);
+
+        Assert.assertEquals(responseForGet.getStatusCode(), 200);
+
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(responseForGet.path("data.name"), user.getName());
+        softAssert.assertEquals(responseForGet.path("data.email"), user.getEmail());
+        softAssert.assertEquals(responseForGet.path("data.status"), user.getStatus());
+        softAssert.assertEquals(responseForGet.path("data.gender"), user.getGender());
+        softAssert.assertAll();
+
+//      after posting user we should delete created user.
+        ApiCall.delete(userID, endPoint);
+    }
+
+    @Test
+    public void deleteUserTest() {
```

Comment:
this method isn't working, it doesn't remove your posted user, just because you didn't mention that user's id on line 58